build.gradle.kts

```diff
@@ -422,9 +424,86 @@ subprojects {
         }
     }
 
+    plugins.withId("me.champeau.gradle.japicmp") {
+        afterEvaluate {
+            // hack to find the current released version of the project
+            val temp: Configuration = project.configurations.create("tempConfig")
+            project.dependencies.add("tempConfig", "io.opentelemetry:opentelemetry-bom:latest.release")
+            val moduleVersion = project.configurations["tempConfig"].resolvedConfiguration.firstLevelModuleDependencies.elementAt(0).moduleVersion
+            project.configurations.remove(temp)
+
+            val userRequestedBase = project.properties["apiBaseVersion"]
+            val baselineVersion = userRequestedBase ?: moduleVersion
+            val newVersion : String? = project.properties["apiNewVersion"] as String?
+            val baselineArtifact: File?
+            var newArtifact: File? = null
+            val oldGroup = project.group
+            try {
+                //this dance is needed due to gradle not allowing more than one version of a dependency.
+                project.group = "virtual_baseline_for_japicmp"
```

Comment:
You mention a dance but I'm still confused, what's the reason for having to set project.group? Don't think it's related to dependency resolution in any way, only publishing

Reply:
I cribbed from here:  https://android.googlesource.com/platform/external/grpc-grpc-java/+/refs/heads/android10-d4-release/build.gradle

I think gradle will only let a module have a single resolved version of a given artifact.

Reply:
(note, I found that code by following several issues about how to implement this particular kind of check in the japicmp project)

Reply:
this issue: https://github.com/melix/japicmp-gradle-plugin/issues/16

Reply:
I see - I think the comment can be something like "Temporarily change the group name because we want to fetch an artifact with the same Maven coordinates as the project, which Gradle would not allow otherwise."